src/libstore/build.cc

```diff
@@ -526,10 +529,19 @@ class UserLock
 
 };
 
-
 UserLock::UserLock()
 {
     assert(settings.buildUsersGroup != "");
+    createDirs(settings.nixStateDir + "/userpool");
+
+    if (findFreeUser()) return;
+
+    printError("waiting for build users");
```

Comment:
Is this really an error? Maybe `warn` would be a better fit? (I found this after a quick grep -- there might be a more suitable function.)

Also (maybe it's just me), but I think it would be nice to see the waiting message every time we check, e.g. put it inside the do-while loop. It would at least give feedback that "Yes, we're still waiting. Sorry this is taking so long."

Reply:
I'm exactly replicating the behavior of `waiting for locks or build slots` message. It's a `printError` too and it only shows once.

Reply:
Ah, OK, that makes sense. (I don't typically run into the "waiting for locks or build slots" message, so it didn't even occur to me.)

Nevermind, then :)

Reply:
I think this merits further discussion. Maybe both of them should be refactored (see https://github.com/NixOS/nix/issues/2069), but I don't think this discussion belongs in this PR :)